collector/zfs_linux.go

```diff
@@ -81,6 +98,50 @@ func (c *zfsCollector) parseProcfsFile(reader io.Reader, fmtExt string, handler
 	return scanner.Err()
 }
 
-func (c *zfsCollector) updatePoolStats(ch chan<- prometheus.Metric) (err error) {
+func (c *zfsCollector) parsePoolProcfsFiles(zpoolPaths []string, handler func(string, zfsSysctl, int)) (err error) {
+
+	for _, zpoolPath := range zpoolPaths {
+		file, err := os.Open(zpoolPath)
+		if err != nil {
+			log.Debugf("Cannot open %q for reading. Is the kernel module loaded?", zpoolPath)
+			err = zfsNotAvailableError
+		}
+		defer file.Close()
```

Comment:
What I did here is just move the loop around the zpool paths back up into the updatePoolStats function. If you think that needs a separate function, I'll need to refactor that in. But I think this way it looks more like the rest of the code in the ZFS exporter (which I hope makes this all easier to follow in the future).